### PR TITLE
fix: `select_any_in` operator - Broken only for attributes

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Using Any In with an attribute of type single_select was broken for attributes only. It fixes it and adds a test to ensure it doesn't break in future.

It isn't a regression. The initial implementation itself had this bug 

[Before fix loom](https://www.loom.com/share/a5bc67c1b12e4743b27d6cdb8d249226)
[After Fix loom](https://www.loom.com/share/bb7d945896de4ef88d7f5de97176eb63)

## Mandatory Tasks (DO NOT REMOVE)
- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
See loom

### CODE REVIEW

File: packages/app-store/routing-forms/TODO.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 - [x] Connect Form Response to Booking
 - [x] Test with non option based attributes and form field input
 - [ ] Switching from Select to MultiSelect in attribute 
-    - [x] It immediately makes the Routing logic to not work. Fix it.
+    - [ ] It immediately makes the Routing logic to not work. Fix it.
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/TODO.md:11]: Compatibility layer is removed.

**Review by zomars — DISMISSED** [review]

**Comment by zomars** [comment]
@hariombalhara we have a type failure. Not sure if related  
image

**Comment by hariombalhara** [comment]
@zomars Fixed it

**Review by Udit-takkar — COMMENTED** [review]
~I added a route which checks if country is "New zealand" and Position is IC1 it should route to team/team1/maybe but it is routing to team/team1/team1-event1~

**Review by Udit-takkar — DISMISSED** [review]

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**